src/pull-request/pull-request.service.ts

```diff
@@ -56,20 +56,16 @@ export class PullRequestService {
       tap(head => keyboardsHead = head),
       switchMap(() => this.gitService.readLastNote(keyboardsRepoPath)),
       map(keyboardsNoteInfo => {
-        if (
-          (new RegExp(`KDO.+${keyboardsNoteInfo.commitSha}$`).exec(singleKbNoteInfo.message) == null ||
-            new RegExp(`KDO.+${singleKbNoteInfo.commitSha}$`).exec(keyboardsNoteInfo.message) == null) &&
-          singleKbNoteInfo.message !== '' && keyboardsNoteInfo.message !== ''
-        ) {
-          throw new Error('Non-linear history. Force-push is not allowed.');
+        if (keyboardsHead !== keyboardsNoteInfo.commitSha && keyboardsNoteInfo.commitSha !== '') {
+          throw new HttpException('Keyboards repo has new changes. This is not allowed.', 409);
```

Comment:
Can you explain the circumstances of this? The keyboards repo will continually get new changes so we need to handle that

Reply:
What I meant here is that the single-keyboard directory (that corresponds to the single-keyboard repo) has new changes.